src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -0,0 +1,390 @@
+ *    <li>
+ *       Second, by {@code @Override}ing the {@link #repeat()} method.
+ *       The code within {@link #repeat()} will run every repetition.
+ */
+public class RepeatingTaskThread extends Thread
+{
+   public static final int REPEAT_INDEFINITELY = -1;
+   public static final double UNLIMITED_FREQUENCY = -1.0;
+
+   private final RunnableThatThrows task;
+   private final ExceptionHandler exceptionHandler;
+   private final Object loopLock = new Object();
+
+   /** Throttler for optionally set loop period/frequency limit */
+   private final Throttler throttler = new Throttler();
+
+   /**
+    * Countdown for number of repetitions to run.
+    * The counter is decremented each time before the repetition.
+    * Once the counter hits 0, the loop is paused.
+    * <ul>
+    *    <li> 0 = pause (don't run the loop until counter value is changed).
+    *    <li> -1 = repeat indefinitely (keep looping until told otherwise).
+    *    <li> N > 0 = run the loop N more repetitions.
+    */
+   private volatile int remainingRepetitions = 0;
+
+   /** Counter for the total number of repetitions completed during the lifetime of this thread. */
+   private long completedRepetitions = 0L;
+
+   /**
+    * Indicates whether this object is destroyed.
+    * The loop will come to a finish when {@code isRunning() == false}.
+    * Does not equate to {@link Thread#isAlive()}, as the thread may take
+    * some time to finish executing after {@code isRunning()} becomes {@code false}.
+    */
+   private volatile boolean running = false;
+
+   /** The optionally set lower limit to the loop period. A negative value indicates no limit */
+   private volatile double loopPeriodLowerLimit = UNLIMITED_FREQUENCY;
+
+   public RepeatingTaskThread(String name)
+   {
+      this(UNLIMITED_FREQUENCY, name);
+   }
+
+   public RepeatingTaskThread(double loopFrequencyLimit, String name)
+   {
+      this(DefaultExceptionHandler.MESSAGE_AND_STACKTRACE, loopFrequencyLimit, name);
+   }
+
+   public RepeatingTaskThread(ExceptionHandler exceptionHandler, String name)
+   {
+      this(exceptionHandler, UNLIMITED_FREQUENCY, name);
+   }
+
+   public RepeatingTaskThread(ExceptionHandler exceptionHandler, double loopFrequencyLimit, String name)
+   {
+      this(null, exceptionHandler, loopFrequencyLimit, name);
+   }
+
+   public RepeatingTaskThread(RunnableThatThrows task, String name)
+   {
+      this(task, UNLIMITED_FREQUENCY, name);
+   }
+
+   public RepeatingTaskThread(RunnableThatThrows task, double loopFrequencyLimit, String name)
+   {
+      this(task, DefaultExceptionHandler.MESSAGE_AND_STACKTRACE, loopFrequencyLimit, name);
+   }
+
+   public RepeatingTaskThread(RunnableThatThrows task, ExceptionHandler exceptionHandler, String name)
+   {
+      this(task, exceptionHandler, UNLIMITED_FREQUENCY, name);
+   }
+
+   public RepeatingTaskThread(RunnableThatThrows task, ExceptionHandler exceptionHandler, double loopFrequencyLimit, String name)
```

Comment:
I would try to remove loopFrequencyLimit from the constructor. You could make setFrequencyLimit return this to allow fluent-style construction.

Reply:
Sometime fluent construction would use withFrequencyLimit for that